openoa/analysis/wake_losses.py

```diff
@@ -465,6 +481,10 @@ def run(
             self.assume_no_wakes_high_ws_LT_corr = assume_no_wakes_high_ws_LT_corr
         if no_wakes_ws_thresh_LT_corr is not None:
             self.no_wakes_ws_thresh_LT_corr = no_wakes_ws_thresh_LT_corr
+        if min_ws_bin_lin_reg is not None:
+            self.min_ws_bin_lin_reg = min_ws_bin_lin_reg
+        if bin_count_thresh_lin_reg is not None:
+            self.bin_count_thresh_lin_reg = bin_count_thresh_lin_reg
```

Comment:
Why not just define the default value for min_ws_bin_lin_reg and bin_count_thresh_lin_reg as default arguments in the run() argument list?

I see two problems with the current implementation:
1. Added complexity (lines 484-487, the use of an attrs field)
2. Unless I'm looking at this wrong - arguments to run() will now "stick" - and become the new default value next time run() is called. This is inconsistent with the documentation which reads "Defaults to 3.0." In fact, it will default to 3.0 or to whichever value was last passed to "run()".

Reply:
So I'm going to push back on this because this relates back to an implementation schematic that's been lingering without feedback for a couple of months. In the new API we're now using the `self.parameter` when accessing instead of just `parameter` as we had done previously. Additionally, we're using a set of attrs validators to check the inputs in many cases to ensure they're valid, so we'd have to then revert back to just manually checking all the inputs.

Would it be helpful to add in a method like the following to allow a resetting of the values when a user wants to clear the slate?

```python
def reset_defaults(self, which: str | list[str] | tuple[str], None = None):
    """"Reset all or a subset of the analysis parameters back to their defaults."""
    # Define the analysis class run parameters
    valid = ["""analysis class run parameters"""]
    
    # If None, set to all run parameterizations
    if which is None:
        which = valid
    
    # Check that all values of which are valid
    ...

    # Reset the selected values back to their defaults
    for attribute in attrs.fields(self):
        if attribute.name in which:
            setattr(self, attribute.name, attribute.default)
```